symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/MessageService.java

```diff
@@ -1,34 +1,128 @@
 package com.symphony.bdk.core.service;
 
+
 import com.symphony.bdk.core.api.invoker.ApiException;
 import com.symphony.bdk.core.auth.AuthSession;
 import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
 import com.symphony.bdk.gen.api.MessagesApi;
+import com.symphony.bdk.gen.api.model.V4ImportResponse;
+import com.symphony.bdk.gen.api.model.V4ImportedMessage;
 import com.symphony.bdk.gen.api.model.V4Message;
 import com.symphony.bdk.gen.api.model.V4Stream;
 
+import com.symphony.bdk.template.api.Template;
+
+import com.symphony.bdk.template.api.TemplateEngine;
+
+import com.symphony.bdk.template.api.TemplateException;
+
 import lombok.extern.slf4j.Slf4j;
 import org.apiguardian.api.API;
 
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.List;
+
 /**
- * PLEASE PLEASE don't review this class !!
+ * Service class for managing messages.
+ * This service can be used to:
+ * <ul>
+ *   <li>send a message to a given stream ID</li>
+ *   <li>get a message given its message ID</li>
+ *   <li>import messages to Symphony</li>
+ * </ul>
  */
 @Slf4j
 @API(status = API.Status.EXPERIMENTAL)
 public class MessageService {
 
   private final MessagesApi messagesApi;
   private final AuthSession authSession;
+  private final TemplateEngine templateEngine;
+  private Template template;
```

Comment:
Looks very risky to have this class property, especially in a multi-thread environment (e.g. SpringBoot)